client/src/main/java/io/pravega/client/stream/TransactionalEventStreamWriter.java

```diff
@@ -26,17 +26,19 @@
      * Start a new transaction on this stream. This allows events written to the transaction be written an committed atomically.
      * Note that transactions can only be open for {@link EventWriterConfig#transactionTimeoutTime}.
      * 
+     * @throws TxnFailedException The Transaction is no longer in state {@link Transaction.Status#OPEN}
      * @return A transaction through which multiple events can be written atomically.
      */
-    Transaction<Type> beginTxn();
+    Transaction<Type> beginTxn() throws TxnFailedException;
 
     /**
      * Returns a previously created transaction.
      * 
      * @param transactionId The result retained from calling {@link Transaction#getTxnId()}
+     * @throws TxnFailedException The Transaction is no longer in state {@link Transaction.Status#OPEN}
```

Comment:
I don't think we want this behavior. 
It seems legitimate to me to do:
```
getTxn(id).checkStatus()
```
In this case I would expect the result to return whatever the status was even if it is not `OPEN`. 
If the code doesn't work this way. we should open an issue.